ai/unet/Model Training.py

```
@@ -0,0 +1,528 @@
+# POST-PROCESSING (to be used in predict.py)
+# ═══════════════════════════════════════════════════════════════════════════
+
+def morphological_postprocess(pred_disc_np, pred_cup_np):
+    """
+    1. Morphological Closing  → fills holes in masks
+    2. Largest Component      → removes isolated artifacts
+    3. Cup ⊂ Disc constraint  → cup must lie inside the disc (anatomy)
+    """
+    kernel = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (7, 7))
+
+    def clean(mask):
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `clean`

(ANN202)